src/server/game/Entities/GameObject/GameObject.h

```diff
@@ -991,7 +983,7 @@ class GameObject : public WorldObject, public GridObject<GameObject>, public Mov
     bool        m_spawnedByDefault;
     uint32       m_cooldownTime;                         // used as internal reaction delay time store (not state change reaction).
     // For traps this: spell casting cooldown, for doors/buttons: reset time.
-    GuidList m_SkillupList;
+    std::unordered_map<ObjectGuid, int32> m_SkillupList;
```

Comment:
Why using int32 instead of uint32, so we don't have to static cast every time? Or are you using int32 for other reason?

Reply:
Yes - look here https://github.com/azerothcore/azerothcore-wotlk/pull/9203/files#diff-386f154bdfb8cad4f9313fa7fb926ed996adbe1449933d53c01f88a666c7ff82R2868